### ISSUE TITLE
Importing Supplied ES template into ES 5.5 results in errors

### ISSUE THREAD

**Issue by alias454** [HASecuritySolutions/VulnWhisperer#1]
#! Deprecation: match_mapping_type [float] is invalid and will be ignored: No field type matched on [float], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: match_mapping_type [byte] is invalid and will be ignored: No field type matched on [byte], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: match_mapping_type [short] is invalid and will be ignored: No field type matched on [short], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: match_mapping_type [integer] is invalid and will be ignored: No field type matched on [integer], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: match_mapping_type [geo_point] is invalid and will be ignored: No field type matched on [geo_point], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: The [string] field is deprecated, please use [text] or [keyword] instead on [source]
#! Deprecation: The [string] field is deprecated, please use [text] or [keyword] instead on [synopsis]
#! Deprecation: The [string] field is deprecated, please use [text] or [keyword] instead on [see_also]
#! Deprecation: The [string] field is deprecated, please use [text] or [keyword] instead on [cve]
#! Deprecation: The [string] field is deprecated, please use [text] or [keyword] instead on [solution]
#! Deprecation: The [string] field is deprecated, please use [text] or [keyword] instead on [@version]
#! Deprecation: The [string] field is deprecated, please use [text] or [keyword] instead on [risk]
#! Deprecation: [omit_norms] is deprecated, please use [norms] instead with the opposite boolean value
{
  "acknowledged": true
}

Current Versions:
elasticsearch v5.5.1
kibana v5.5.1
logstash v5.5.1

**Comment by austin-taylor** [comment]
Hi Alias454, did you update the template?

**Comment by alias454** [comment]
Hi @austin-taylor I used the one located here https://github.com/austin-taylor/VulnWhisperer/tree/master/elasticsearch if that is what you are meaning. or do you mean did I make other changes?

**Comment by austin-taylor** [comment]
Looks like we may need to create a template for 5.5

**Comment by austin-taylor** [comment]
Mind trying to apply the recommendations above to the template and let me know if it works?

**Comment by alias454** [comment]
Sure, I got down to this this so far. and I have a little reading to do.
 
#! Deprecation: match_mapping_type [float] is invalid and will be ignored: No field type matched on [float], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: match_mapping_type [byte] is invalid and will be ignored: No field type matched on [byte], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: match_mapping_type [short] is invalid and will be ignored: No field type matched on [short], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: match_mapping_type [integer] is invalid and will be ignored: No field type matched on [integer], possible values are [object, string, long, double, boolean, date, binary]
#! Deprecation: match_mapping_type [geo_point] is invalid and will be ignored: No field type matched on [geo_point], possible values are [object, string, long, double, boolean, date, binary]
{
  "acknowledged": true
}

Adding for reference: https://www.elastic.co/blog/strings-are-dead-long-live-strings

**Comment by austin-taylor** [comment]
I'm working on a blog to discuss installation details right now. I will make sure to include this in there.

**Comment by austin-taylor** [comment]
Did you have any trouble getting the visualizations to import?

**Comment by alias454** [comment]
I can apply this template without getting any depreciation warnings. However, I still get errors when importing the visualizations.

```
PUT _template/logstash-nessus
{
    "order":0,
    "template":"logstash-nessus-*",
    "settings":{
        "index":{
            "refresh_interval":"5s"
        }
    },
    "mappings":{
        "_default_":{
            "dynamic_templates":[
                {
                    "message_field":{
                        "path_match":"message",
                        "mapping":{
                            "norms":false,
                            "type":"text"
                        },
                        "match_mapping_type":"string"
                    }
                },
                {
                    "string_fields":{
                        "mapping":{
                            "index":"true",
                            "norms":false,
                            "type":"text",
                            "fields":{
                                "raw":{
                                    "ignore_above":256,
                                    "index":"false",
                                    "type":"keyword"
                                }
                            }
                        },
                        "match_mapping_type":"string",
                        "match":"*"
                    }
                },
                {
                    "date_fields":{
                        "mapping":{
                            "type":"date",
                            "doc_values":true
                        },
                        "match_mapping_type":"date",
                        "match":"*"
                    }
                },
                {
                    "ip_address_fields":{
                        "mapping":{
                            "type":"ip"
                        },
                        "match":"*_ip"
                    }
                }
            ],
            "properties":{
                "@timestamp":{
                    "type":"date"
                },
                "plugin_id":{
                    "type":"integer",
                    "fields":{
                        "raw":{
                            "index":"false",
                            "type":"keyword"
                        }
                    }
                },
                "last_updated":{
                    "type":"date",
                    "doc_values":true
                },
                "geoip":{
                    "dynamic":true,
                    "properties":{
                        "ip":{
                            "type":"ip"
                        },
                        "latitude":{
                            "type":"half_float"
                        },
                        "location":{
                            "type":"geo_point"
                        },
                        "longitude":{
                            "type":"half_float"
                        }
                    }
                },
                "risk_score":{
                    "type":"float"
                },
                "synopsis":{
                    "index":"false",
                    "type":"text",
                    "fields":{
                        "raw":{
                            "index":"false",
                            "type":"keyword"
                        }
                    }
                },
                "see_also":{
                    "index":"false",
                    "type":"text",
                    "fields":{
                        "raw":{
                            "index":"false",
                            "type":"keyword"
                        }
                    }
                },
                "cve":{
                    "index":"false",
                    "type":"keyword",
                    "fields":{
                        "raw":{
                            "index":"false",
                            "type":"keyword"
                        }
                    }
                },
                "solution":{
                    "index":"false",
                    "type":"keyword",
                    "fields":{
                        "raw":{
                            "index":"false",
                            "type":"keyword"
                        }
                    }
                },
                "port":{
                    "type":"integer"
                },
                "@version":{
                    "index":"false",
                    "type":"keyword"
                },
                "risk":{
                    "index":"true",
                    "type":"keyword",
                    "fields":{
                        "raw":{
                            "index":"false",
                            "type":"keyword"
                        }
                    }
                },
                "cvss":{
                    "type":"float",
                    "fields":{
                        "raw":{
                            "index":"false",
                            "type":"keyword"
                        }
                    }
                }
            }
        }
    },
    "aliases":{ }
}
```

**Comment by alias454** [comment]
Here is a screenshot of the visualization errors.
![kibana_errors](https://user-images.githubusercontent.com/10768415/28992605-94239580-7966-11e7-8bee-685c32c560c9.PNG)

**Comment by austin-taylor** [comment]
Ah, yes you are missing the plugins. I am including that in the blog post. For example, health metric can be found at: https://github.com/clamarque/kibana_health_metric_vis and pre_alert is used for swimlanes: https://github.com/prelert/kibana-swimlane-vis/releases/

I'm working on a dockerized version of this which should dramatically decrease installation complexity. Also will have link to all visualization plugins in blog

**Comment by alias454** [comment]
I installed the swimlane plugin without issue but the health_metric plugin is not yet updated to be compatible with kibana 5.5.1. I opened an issue with the maintainer and I am trying to sort it out myself. I got it to compile cleanly and install but having issues with the display. I think with a little more fiddling around, I can get it to work.

I still had to make some adjustments to the visualizations because of some field names but I am pretty close to having it working.
